src/parsers/avParser/index.ts

```diff
@@ -0,0 +1,281 @@
+import * as bluebird from 'bluebird';
+import cheerio = require('cheerio');
+import * as _ from 'lodash';
+import * as request from 'request-promise';
+import { AV_URL, codeErrors, proxy } from '../../config/config';
+import { IAvMark } from '../../interfaces/parserInterface';
+import { ParserError } from '../../utils/errors/';
+
+// tslint:disable-next-line:no-var-requires
+const Agent = require('socks5-https-client/lib/Agent');
+
+export const getMarks = async () => {
+  let response: any;
+  try {
+    response = await request.get({
+      agentClass: Agent,
+      agentOptions: {
+        socksHost: 'localhost',
+        socksPort: 9060
+      },
+      url: AV_URL
+    });
+    const $ = cheerio.load(response, {
+      normalizeWhitespace: true,
+      xmlMode: true
+    });
+    const marksURLs: string[] = $('.brandslist')
+      .find('a')
+      .map(function() {
+        return $(this).attr('href');
+      })
+      .get()
+      .slice(0, -1);
+    const marksNames: string[] = $('.brandslist')
+      .find('span')
+      .map(function() {
+        return $(this).text();
+      })
+      .get();
+
+    const marks: any[] = _.map(marksNames, (name, index) => ({ name, url: marksURLs[index] }));
+    return marks;
+  } catch (e) {
+    throw new ParserError(codeErrors.AV_PARSE_ERROR);
+  }
+};
+
+export const getModels = async (marks: IAvMark[]) => {
+  const get = (mark: any) => {
+    return new Promise(resolve => {
+      request
+        .get({
+          agentClass: Agent,
+          agentOptions: {
+            socksHost: 'localhost',
+            socksPort: 9060
+          },
+          uri: mark.url
+        })
+        .then(response => {
+          const $ = cheerio.load(response, {
+            normalizeWhitespace: true,
+            xmlMode: true
+          });
+          const modelsURLs: string[] = $('.brandslist')
+            .find('a')
+            .map(function() {
+              return $(this).attr('href');
+            })
+            .get();
+          const modelsNames: string[] = $('.brandslist')
+            .find('span')
+            .map(function() {
+              return $(this).text();
+            })
+            .get();
+          const transformedModels: any[] = _.map(modelsNames, (name, index) => ({
+            name,
+            url: modelsURLs[index]
+          }));
+          resolve({ [mark.name]: transformedModels });
+        });
+    });
+  };
+
+  let models: any[] = [];
+  const date = new Date();
+  for (let i = 0; i < marks.length; i += 3) {
+    const currentMarks = _.slice(marks, i, i + 3);
+    models = _.concat(
+      models,
+      await Promise.all(_.slice(
```

Comment:
Please either make it more readable or add some comments. Right now it's not clear what happens here.